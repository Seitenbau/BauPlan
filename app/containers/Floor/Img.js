import styled from 'styled-components';

import NormalImg from 'components/Img';

const Img = styled(NormalImg)`
  width: ${(props) => props.theme.plans.image.width}vw;
  height: auto;
  margin: 0 auto;
`;

export default Img;
