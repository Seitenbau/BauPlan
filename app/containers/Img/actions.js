import {
  DISPATCH_CALCULATION,
} from './constants';

export function dispatchCalculation(data) {
  return {
    type: DISPATCH_CALCULATION,
    data,
  };
}
