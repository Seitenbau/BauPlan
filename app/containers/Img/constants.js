export const DEFAULT_ACTION = 'app/Img/DEFAULT_ACTION';
export const CALCULATION_DONE = 'app/Img/CALCULATION_DONE';
