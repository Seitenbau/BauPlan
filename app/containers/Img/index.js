/**
*
* Img
*
*/

import React, { PropTypes } from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import {
  makeSelectUiSize,
} from './selectors';
import {
  dispatchCalculation,
} from './actions';

class Img extends React.PureComponent {
  constructor(props) {
    super(props);
    this.doCalculation = this.doCalculation.bind(this);
  }
  shouldComponentUpdate(nextProps) {
    return this.props['width-height'] !== nextProps['width-height'];
  }
  componentWillUpdate() {
    this.props.dispatchCalculation({
      scale: this.componentCalculateScaleFactor(),
      id: this.props.id,
    });
  }
  doCalculation() {
    this.props.dispatchCalculation({
      scale: this.componentCalculateScaleFactor(),
      id: this.props.id,
    });
  }
  componentCalculateScaleFactor() {
    return (this.img.offsetWidth / this.img.naturalWidth) * this.props.mapScaleFactor;
  }
  render() {
    return (
      <img ref={(ref) => { this.img = ref; }} onLoad={this.doCalculation} src={this.props.src} alt={this.props.alt} className={this.props.className} />
    );
  }
}

Img.propTypes = {
  alt: PropTypes.string.isRequired,
  src: PropTypes.string.isRequired,
  className: PropTypes.string,
  dispatchCalculation: PropTypes.func,
  'width-height': PropTypes.string,
  mapScaleFactor: PropTypes.number,
  id: PropTypes.string,
};

function mapDispatchToProps(dispatch) {
  return {
    dispatchCalculation: (interval) => dispatch(dispatchCalculation(interval)),
  };
}

const mapStateToProps = createStructuredSelector({
  'width-height': makeSelectUiSize(),
});

export default connect(mapStateToProps, mapDispatchToProps)(Img);
