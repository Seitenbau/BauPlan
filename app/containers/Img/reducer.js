/*
 *
 * Plans reducer
 *
 */

import { Map, fromJS } from 'immutable';
import {
  DEFAULT_ACTION,
  CALCULATION_DONE,
} from './constants';

const initialState = fromJS({
  scale: Map(),
});

function imgReducer(state = initialState, action) {
  switch (action.type) {
    case DEFAULT_ACTION:
      return state;
    case CALCULATION_DONE:
      return state.set('scale', state.get('scale').merge(Map([[action.data.key, action.data.scale]])));
    default:
      return state;
  }
}

export default imgReducer;
