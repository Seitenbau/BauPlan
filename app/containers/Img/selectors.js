import { createSelector } from 'reselect';
import { selectUiEventProviderDomain } from '../UiEventProvider/selectors';
/**
 * Direct selector to the plans state domain
 */
const selectImgDomain = () => (state) => state.get('img');

const makeSelectUiSize = () => createSelector(
  selectUiEventProviderDomain(),
  (substate) => substate.get('width-height')
);

export {
  selectImgDomain,
  makeSelectUiSize,
};
