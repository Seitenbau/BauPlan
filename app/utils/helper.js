import React from 'react';
import { ThemeProvider } from 'styled-components';
import system from './System';
import theme from './theme.json';
// FIXME: why no import allowed?  import { shallow } from 'enzyme';

import {
  REM_BASE,
 } from './constants';


export function importAll(r) {
  const images = {};
  // eslint-disable-next-line no-return-assign
  r.keys().map((item) =>
    images[item.replace('./', '')] = r(item)
  );
  return images;
}

export function rem(pix) {
  const multi = pix.toString().split(' ');
  if (multi.length > 1) {
    return multi.map((p) => rem(p));
  }
  return `${(parseFloat(pix / REM_BASE, 10))}rem`;
}

export function toTheme(theme, systemStyles) {
  return Object.assign({}, theme, { systemStyles });
}

/*
  percent = 0.x -> shade
  percent = -0.x -> darken
  https://stackoverflow.com/a/13542669/5871835
*/
export function shadeColor(color, percent) {
    const f = parseInt(color.slice(1), 16),
    const t = percent < 0 ? 0 : 255,
    const p = percent < 0 ? percent * -1 : percent,
    const R = f >> 16,
    const G = f >> 8 & 0x00FF,
    const B = f & 0x0000FF;

    return "#"+(0x1000000 +
              (Math.round((t - R) * p) + R) * 0x10000 +
              (Math.round((t - G) * p ) + G) * 0x100 +
              (Math.round((t - B) * p ) + B))
              .toString(16).slice(1);
}

export function prefixNumber(number) {
  if(number <= 9) {
    return `#00${number}`;
  } else if(number <= 99) {
    return `#0${number}`
  }
  return `#${number}`
}

export const shallowWithTheme = (children, options) => {
  const wrapper = shallow(<ThemeProvider theme={theme}>{children}</ThemeProvider>, options)
  const instance = wrapper.root.instance()
  return wrapper.shallow({ context: instance.getChildContext() })
}
