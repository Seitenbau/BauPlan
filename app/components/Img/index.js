/**
*
* Img
*
*/

import React, { PropTypes } from 'react';
import { importAll } from 'utils/helper';
import images from 'utils/Images';
// import styled from 'styled-components';


function Img(props) {
  console.log(images);
  return (
    <img src={images[props.src]} alt={props.src.split('.')[0]} className={props.className} />
  );
}

Img.propTypes = {
  src: PropTypes.string.isRequired,
  className: PropTypes.string,
};

export default Img;
