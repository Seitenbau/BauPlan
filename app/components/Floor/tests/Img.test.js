import React from 'react';
import { shallowWithTheme } from 'utils/helper';

import Img from '../Img';

describe('<Img />', () => {
  const cut = shallowWithTheme(<Img src="floor.svg"></Img>);
  it('Expect to have a name', () => {
    expect(cut.props.src).not().toBeDefined();
  });
});
