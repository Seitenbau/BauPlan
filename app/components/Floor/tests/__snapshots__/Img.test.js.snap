exports[`<Img /> Expect to render correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Img
    className="sc-bdVaJa bIetak"
    src="floors.svg" />,
  "nodes": Array [
    <Img
      className="sc-bdVaJa bIetak"
      src="floors.svg" />,
  ],
  "options": Object {
    "context": Object {
      "__styled-components__": [Function],
    },
  },
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {
        "__styled-components__": [Function],
      },
      "_currentElement": <Styled(undefined)
        src="floors.svg" />,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StyledComponent {
        "_reactInternalInstance": [Circular],
        "attrs": Object {},
        "context": Object {
          "__styled-components__": [Function],
        },
        "props": Object {
          "src": "floors.svg",
        },
        "refs": Object {},
        "state": Object {
          "generatedClassName": "bIetak",
          "theme": Object {
            "colors": Object {
              "backgroundColor": "#F2F5F6",
              "black": "#FFFFFF",
              "primary": "#022032",
              "secondary": "#00FF00",
              "white": "#00000",
            },
            "plans": Object {
              "backgroundColor": "#FFFFFF",
              "color": "#000000",
              "floorName": Object {
                "font": "Helvetica",
                "size": 24,
              },
              "image": Object {
                "padding": "3vh 5vw 3vh 5vw",
                "width": 50,
              },
              "marginBottom": 40,
              "roomName": Object {
                "font": "Helvetica",
                "size": 16,
              },
              "width": 60,
            },
            "systemStyles": Object {
              "name": "Bauplan",
              "objects": Object {
                "table": Object {
                  "color": "#d3d3d3",
                  "name": "Table",
                  "size": Array [
                    1.6,
                    0.8,
                  ],
                },
              },
              "paths": Object {
                "images": "utils/images/",
              },
            },
          },
        },
        "unsubscribe": [Function],
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Img
          className="sc-bdVaJa bIetak"
          src="floors.svg" />,
        "_debugID": 4,
        "_renderedOutput": <Img
          className="sc-bdVaJa bIetak"
          src="floors.svg" />,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Styled(undefined)
    src="floors.svg" />,
}
`;
